### DISCUSSION TITLE
How to save the improvements made from extrinsic camera training?

### DISCUSSION THREAD

**Discussion by claudiomla** [NVlabs/instant-ngp#565] [Q&A]
Hello, could you please tell me how can I save the transformation matrices of the cameras that change when extrinsic training is performed?
I followed the steps in the "INERF" discussion, and it works!!!! Thanks!!!. But I haven't seen the values of the transformation matrices change in the "transforms.json" when I run the extrinsic training, or are they saved somewhere else?
Note: When I do the extrinsic training with the --gui enabled, I see how the camera positions change, but I don't get those changes saved externally.
Thanks

**Comment by Tom94** [comment]
There's no directly functionality for this -- I'm afraid you'll have to wire up Python bindings or a file dump yourself.

**Reply by claudiomla** [reply]
Don't worry, it's ok, I already found where I can get the transformation matrix testbed.nerf.training.get_camera_extrinsics(index) 
Only one thing, which is the order of images that return this function? It has the order given in the transforms.json?
And another question, the matrix returned are very similar but not equals to the matrixes inside of transforms.json, what is the correct?
Thanks! Regards

**Reply by claudiomla** [reply]
I reply my questions
1) The order of the index for testbed.nerf.training.get_camera_extrinsics(index) is in function of name string order, it's not the order of the images inside of transforms.json [0] -> "1.jpg" [1] -> "10.jpg" [2]-> "11.jpg" .... 
2) The transformation matrixes are equal if I don't extrinsic training.
